pkg/gen/node-templates/provider.ts.mustache

```diff
@@ -166,12 +166,13 @@ export namespace yaml {
 
         const kind = obj["kind"];
         const apiVersion = obj["apiVersion"];
-        if (kind == null || apiVersion == null) {
+        const meta = obj["metadata"];
+        if (kind == null || apiVersion == null || meta == null) {
             return null;
         }
 
-        let id: string = obj["metadata"]["name"];
-        const namespace = obj["metadata"]["namespace"] || undefined;
+        let id: string = meta["name"];
+        const namespace = meta["namespace"] || undefined;
```

Comment:
I'm assuming the empty string is not a legal namespace?

Reply:
It is. It means `"default"`.

Reply:
gotcha.